openapi/types.go

```diff
@@ -81,22 +90,38 @@ func DataTypeFromGo(t reflect.Type) DataType {
 		return TypeBoolean
 	case reflect.String:
 		return TypeString
-		// Handle known unsupported types.
-	case reflect.Func, reflect.Chan, reflect.Uintptr, reflect.UnsafePointer, reflect.Interface, reflect.Complex64, reflect.Complex128:
+	case reflect.Map, reflect.Struct, reflect.Array, reflect.Slice:
+		return TypeComplex
+	default:
+		// reflect.Func, reflect.Chan, reflect.Uintptr, reflect.UnsafePointer,
+		// reflect.Interface, reflect.Complex64, reflect.Complex128, ...
 		return TypeUnsupported
 	}
-	// Complex types:
-	// * reflect.Struct
-	// * reflect.Map
-	// * reflect.Slice
-	// * reflect.Array
-	return TypeComplex
 }
 
+// stringToType converts val to t's type and return the new value.
 func stringToType(val string, t reflect.Type) (interface{}, error) {
+	// Compare type to know Gokang types.
+	// IT MUST BE EXECUTED BEFORE swithing over
+	// primitives because a time.Duration is itself
+	// a int64.
+	if t.AssignableTo(tofTime) {
+		return time.Parse(time.RFC3339, val)
+	}
+	if t.AssignableTo(tofDuration) {
+		return time.ParseDuration(val)
+	}
 	switch t.Kind() {
 	case reflect.Bool:
-		return strconv.ParseBool(val)
+		// ParseBool returns an error if the value
+		// is invalid and cannot be converted to a
+		// boolean. We assume that invalid values
+		// are always falsy.
+		v, err := strconv.ParseBool(val)
+		if err != nil {
```

Comment:
Actually, if you're gonna ignore the error, you can return it as is:

```go
v, _ := strconv.ParseBool(val)
return v, nil
```

Since `strconv.ParseBool` will just return `false` in case of an error.